crates/iceberg-catalog/src/catalog/views/commit.rs

```diff
@@ -108,8 +111,13 @@ pub(crate) async fn commit_view<C: Catalog, A: Authorizer + Clone, S: SecretStor
     // serialize body before moving it
     let body = maybe_body_to_json(&request);
 
-    let requested_update_metadata = build_new_metadata(request, before_update_metadata)?;
+    let (requested_update_metadata, delete_old_location) = build_new_metadata(
+        request,
+        before_update_metadata,
+        &before_update_view_location,
+    )?;
 
+    let view_location = parse_view_location(requested_update_metadata.location())?;
```

Comment:
If the new location != old location, we must call `storage_profile.require_allowed_location(&location)?`